task/upload.go

```diff
@@ -264,6 +323,29 @@ func toTranscodeFileTaskOutput(callback *clients.CatalystCallback) (data.Transco
 	return res, nil
 }
 
+func toClipOutput(callback *clients.CatalystCallback) (data.ClipTaskOutput, error) {
+	var res data.ClipTaskOutput
+
+	res.RequestID = callback.RequestID
+
+	if len(callback.Outputs) < 1 {
+		return res, fmt.Errorf("invalid video outputs: %v", callback.Outputs)
+	}
+	// we expect only one output
+	o := callback.Outputs[0]
+
+	bu, p, err := parseUrlToBaseAndPath(o.Manifest)
+	if err != nil {
+		return res, err
+	}
+	res.BaseUrl = bu
+	if len(o.Videos) > 0 {
```

Comment:
So clips may or may not have HLS, is that it? Does that mean assets can also have only an Mp4 now but not HLS? Or is it the other way around?

IOW what happens if there is no videos here? We return a base URL that points to a folder but there are no files there? Shouldn't that be an error?

Reply:
We changed everything with output locations and now the clip task output is going to go with the same output locations of VOD, + the clip source